ml-agents/mlagents/trainers/training_status.py

```diff
@@ -103,6 +107,56 @@ def set_parameter_state(category: str, key: StatusType, value: Any) -> None:
         """
         GlobalTrainingStatus.saved_state[category][key.value] = value
 
+    @staticmethod
+    def append_to_parameter_state(category: str, key: StatusType, value: Any) -> None:
+        """
+        Appends an arbitrary-named parameter in the global saved state.
+        :param category: The category (usually behavior name) of the parameter.
+        :param key: The parameter, e.g. lesson number.
+        :param value: The value.
+        """
+        GlobalTrainingStatus.saved_state[category][key.value].append(value)
+
+    @staticmethod
+    def remove_checkpoint(checkpoint: Dict[str, Any]) -> None:
```

Comment:
I don't think checkpoint manager should go in GlobalTrainingStatus. Maybe a separate CheckpointManager class that under the hood calls GlobalTraining Status?

Reply:
Agreed -- may also be reasonable for the logic to live within Trainer, but not here.